internal/pkg/cli/pipeline_init.go

```diff
@@ -158,8 +157,23 @@ func (o *initPipelineOpts) Validate() error {
 		return err
 	}
 
+	// URL flag and branch flag must both be set or not.
+	if URL, branch := o.repoURL == "", o.repoBranch == ""; URL != branch {
```

Comment:
Why do they have to specify branch flag when using the repo URL? Like why not set default branch for them?

Reply:
I guess this is to be consistent with the guided experience? Like with guided experience, you have to select a branch from the prompts

Reply:
hmm then can we not throw an error but ask them for the branch name if branch flag is not set?

Reply:
I think when we have multiple pipeline support, users have to be much more intentional about their branches. Right now the default branch name just gets stuck in the pipeline manifest and the users may not even notice until their pipeline build fails (like in #2105, and my own experience). 

I like the idea of asking for the flag if it's not set! Is that breaking the paradigm, though, doing an `Ask` within `Validate`? I'm not seeing any other "must be specified with" situations besides in `task run` and that seems to be a different story.

Reply:
I think we usually `Ask` for properties that are not specified in flags. That is in `Ask()`, we check if `o.branch == nil`, and if it is, we `askForBranch`.
`task run` doesn't have such situations though, it only has "must not be specified with" situations 🤔

Reply:
Right...that is my concern. If someone passes in a branch/URL but not the other, we're performing an `Ask()` type of behavior by prompting them for the other, rather than erroring out. Which may be fine....

For `task run` I added the requirement of `platform-os` and `platform-arch` both being required if one is.

Reply:
Changed code to ask (via selector) for missing branch/URL if the other is passed in.
Only the branches of the passed in URL are listed in the selector; only the URLs that contain the passed in branch are listed.
If there is only one option for either branch or URL, that one automatically is set, and a message is logged. (Same code for `Ask()`.

Reply:
Changed again to delay the asking until `Ask()`